Sources/swift-doctest/main.swift

```diff
@@ -81,6 +90,22 @@ struct SwiftDocTest: ParsableCommand {
 
         let consolidatedReport = Report.consolidation(of: reports)
         standardOutput.write(consolidatedReport.description.data(using: .utf8)!)
+        let failureList = consolidatedReport.results.filter({ result -> Bool in
+            do {
+                return try !result.get().ok
+            } catch {
+
+                return true
+            }
+        })
+        if (failureList.count > 0) {
+            // return a non-zero result code so that CI systems will react appropriately
+            #if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
+            Darwin.exit(Int32(failureList.count))
+            #elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android) || os(Cygwin) || os(Haiku)
+            Glibc.exit(Int32(failureList.count))
+            #endif
+        }
```

Comment:
```suggestion
            // Return a non-zero result code if any tests failed
            #if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
            Darwin.exit(EXIT_FAILURE)
            #elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android) || os(Cygwin) || os(Haiku)
            Glibc.exit(EXIT_FAILURE)
            #endif
        }
```

Reply:
I don't think we should signal with an exit status other than success or failure. Passing the number of failed tests has the potential of signaling a status in excess of `255` (which may cause problems of some platforms) and deprives us of the ability to designate status codes for certain kinds of failures.